src/PowerShell/Microsoft.WinGet.DSC/Microsoft.WinGet.DSC.psm1

```diff
@@ -0,0 +1,228 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+using namespace System.Collections.Generic
+
+try
+{
+    # Load all non-test .ps1 files in the script's directory.
+    Get-ChildItem -Path $PSScriptRoot\* -Filter *.ps1 -Exclude *.Tests.ps1 -Recurse | ForEach-Object { Import-Module $_.FullName }
+} catch
+{
+    $e = $_.Exception
+    while ($e.InnerException)
+    {
+        $e = $e.InnerException
+    }
+
+    if (-not [string]::IsNullOrWhiteSpace($e.Message))
+    {
+        Write-Host $e.Message -ForegroundColor Red -BackgroundColor Black
+    }
+}
+
+#region classes
+
+# Literally the same as Microsoft.Management.Deployment.PackageCatalogReference.
+# Using that type in a DSC resource it fails with MethodInvocationException: Exception calling
+# "ImportClassResourcesFromModule" with "4" argument(s): "The DSC resource 'PackageCatalogReference' has no default constructor."
+# TODO: figure that out.
+class WinGetSource

```

Comment:
>WinGetSource [](http://example.com/codeflow?start=6&length=12)

Update this to use a similar thing like the group policy source style, which is kind of the same but with two more params or just a hashtable?